### PR TITLE
DevWs for TechniSat Digital GmbH containing containing TechniSat Double-Switch

### PR DESCRIPTION
**Pull request by SmartThings**

### CODE REVIEW

**Comment by NArlt** [comment]
To add child DTH new PR created: #66242

**Comment by MWierzbinskaS** [comment]
@NArlt Sorry for the late reply. Could You please check if this child device can be integrated with one of existing child DTH's? e.g. https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/master/devicetypes/smartthings/child-metering-switch.src/child-metering-switch.groovy

**Comment by NArlt** [comment]
Hi @MWierzbinskaS ,
I tried to use the existing child DTH but not successfully. The Problem is our main DTH is in our own namespace and I couldn't find a way to use a DTH from another namespace. If there is a way I will change to the DTH you mentioned.

File: devicetypes/technisat/technisat-series-switch.src/technisat-series-switch.groovy [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+	}
+}
+
+private createChild() {
+	
+	log.debug "createChild componentLabel: ${componentLabel}"
+	try {
+		String dni = "${device.deviceNetworkId}-ep2"
+        def componentLabel = "${device.displayName[0..-2]}2"
+		addChildDevice("TechniSat Series switch child", dni, device.getHub().getId(),
```
Comment (greens) [inline devicetypes/technisat/technisat-series-switch.src/technisat-series-switch.groovy:58]: I don't see this DTH anywhere. You'd need to include it in this PR.

File: devicetypes/technisat/technisat-series-switch.src/technisat-series-switch.groovy [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+		}
+	}
+}
+
+private createChild() {
+	
+	log.debug "createChild componentLabel: ${componentLabel}"
+	try {
+		String dni = "${device.deviceNetworkId}-ep2"
+        def componentLabel = "${device.displayName[0..-2]}2"
```
Comment (greens) [inline devicetypes/technisat/technisat-series-switch.src/technisat-series-switch.groovy:57]: nit: spacing

File: devicetypes/technisat/technisat-series-switch.src/technisat-series-switch.groovy [hunk 3]
```diff
@@ -0,0 +1,440 @@
...
+ *	Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *	on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *	for the specific language governing permissions and limitations under the License.
+ *
+ */
+import groovy.json.JsonOutput
+
+metadata  {
+	definition (name: "TechniSat Series switch", namespace: "TechniSat", author: "TechniSat", vid:"generic-switch-power-energy",
+				mnmn: "SmartThings", runLocally: true, minHubCoreVersion: '000.017.0012',
```
Comment (greens) [inline devicetypes/technisat/technisat-series-switch.src/technisat-series-switch.groovy:18]: `runLocally` should be false. the need to address on commands to endpoint 1 I believe will make this not work locally. you could also just remove the `runLocally`, `minHubCoreVersion`, and `executesCommandsLocally` flags.

File: devicetypes/technisat/technisat-series-switch.src/technisat-series-switch.groovy [hunk 4]
```diff
@@ -0,0 +1,440 @@
...
+ *
+ *	Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *	on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *	for the specific language governing permissions and limitations under the License.
+ *
+ */
+import groovy.json.JsonOutput
+
+metadata  {
+	definition (name: "TechniSat Series switch", namespace: "TechniSat", author: "TechniSat", vid:"generic-switch-power-energy",
```
Comment (greens) [inline devicetypes/technisat/technisat-series-switch.src/technisat-series-switch.groovy:17]: capitalization of your `namespace` needs to match the capitalization of the folder under `devicetypes` I believe

File: devicetypes/technisat/technisat-series-switch.src/technisat-series-switch.groovy (hunk 1 above)
Comment (NArlt) [reply devicetypes/technisat/technisat-series-switch.src/technisat-series-switch.groovy:58]: This request is created via DevWs and it wasn't possible to add a 2nd DTH for 1 product there. So I created the publish request with only the 'main' DTH and immediately after this a 2nd PR including the child DTH which I already linked here (#66242)

File: devicetypes/technisat/technisat-series-switch.src/technisat-series-switch.groovy (hunk 4 above)
Comment (NArlt) [reply devicetypes/technisat/technisat-series-switch.src/technisat-series-switch.groovy:17]: This is the 4th DTH I commit with this namespace. The folder name was auto-created with the first DTH. It seems to work for the other 3, already approved, DTHs. But if it is really a problem I have to change the other DTHs as well

**Comment by NArlt** [comment]
I made the requested changes and committed to PR #66242

File: devicetypes/technisat/technisat-series-switch.src/technisat-series-switch.groovy (hunk 4 above)
Comment (NArlt) [reply devicetypes/technisat/technisat-series-switch.src/technisat-series-switch.groovy:17]: After checking some other DTHs is also noticed, the namespace not always matches the folder name. For example in folder  fibargroup the namespace "FibarGroup" is used, or in drzwave the namespace "DrZWave" is used.

**Comment by greens** [comment]
I'll be following up on that PR, since it contains both DTHs.

**Comment by k-morawski** [comment]
I am closing this pull request.

**Pull request closed without merging**